### PR TITLE
[API]delete address book

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially #11914 ( this needs to be merged first)
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Address.xml [hunk 1]
```diff
@@ -30,6 +30,10 @@
                 <attribute name="method">GET</attribute>
                 <attribute name="path">/shop/addresses/{id}</attribute>
             </itemOperation>
+            <itemOperation name="shop_delete">
+                <attribute name="method">DELETE</attribute>
+                <attribute name="path">/shop/addresses/{id}</attribute>
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Address.xml:35]: we have address book only on shop context są instead custom `path` you can add parameter `route_prefix`

File: src/Sylius/Behat/Context/Api/Shop/AddressContext.php [hunk 2]
```diff
@@ -111,4 +132,19 @@ private function addressBookHasAddress(array $addressBook, AddressInterface $add
...
+    private function getAddressIdFromAddressBookByFullName(string $fullName): ?string
+    {
+        Assert::notNull($fullName);
+        $fullNameArray = explode(' ',$fullName);
+
+        $addresses = $this->responseChecker->getCollection($this->client->getLastResponse());
+        /** @var AddressInterface $address */
+        foreach ($addresses as $address){
+            if ($fullNameArray[0] === $address['firstName'] && $fullNameArray[1] === $address['lastName']) {
+                $addressIriStringArray = explode('/', $address['@id']);
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Shop/AddressContext.php:145]:
```suggestion
                $addressIriArray = explode('/', $address['@id']);
```

File: features/account/customer_account/address_book/deleting_address_created_after_checkout.feature [hunk 3]
```diff
@@ -25,3 +25,18 @@ Feature: Removing an address from my book
...
+        And I am at the checkout addressing step
+        And my billing address is fulfilled automatically through default address
+        When I specify the first and last name as "Mike Ross" for billing address
+        And I complete the addressing step
+        And I proceed with "Free" shipping method and "Offline" payment
+        And I confirm my order
+        And I browse my address book
+        And I delete the "Mike Ross" address
+        Then I should be notified that the address has been successfully deleted
+        Then I browse my address book
```
Comment (AdamKasp) [inline features/account/customer_account/address_book/deleting_address_created_after_checkout.feature:41]:
```suggestion
        And I browse my address book
```
And i think you can merge these two scenarioes in one.

File: features/account/customer_account/address_book/deleting_address_created_after_checkout.feature [hunk 4]
```diff
@@ -25,3 +25,18 @@ Feature: Removing an address from my book
         And I delete the "Mike Ross" address
         Then I should be notified that the address has been successfully deleted
         And I should have a single address in my address book
+
+    @api
+    Scenario: Deleting address created after placing an order
```
Comment (GSadee) [inline features/account/customer_account/address_book/deleting_address_created_after_checkout.feature:30]: Why did you add another scenario?

File: features/account/customer_account/address_book/deleting_address_created_after_checkout.feature [hunk 5]
```diff
@@ -25,3 +25,18 @@ Feature: Removing an address from my book
...
+        And I am at the checkout addressing step
+        And my billing address is fulfilled automatically through default address
+        When I specify the first and last name as "Mike Ross" for billing address
+        And I complete the addressing step
+        And I proceed with "Free" shipping method and "Offline" payment
+        And I confirm my order
+        And I browse my address book
+        And I delete the "Mike Ross" address
+        Then I should be notified that the address has been successfully deleted
+        And I browse my address book
```
Comment (GSadee) [inline features/account/customer_account/address_book/deleting_address_created_after_checkout.feature:41]:
```suggestion
```
Getting address book should be hidden in Then step I guess

File: src/Sylius/Behat/Context/Api/Shop/AddressContext.php [hunk 6]
```diff
@@ -89,9 +90,29 @@ public function iShouldNotSeeTheAddressAssignedTo(string $fullName): void
      */
     public function thereShouldBeNoAddresses(): void
     {
+        $this->client->index();
+
         Assert::same(count($this->responseChecker->getCollection($this->client->getLastResponse())), 0);
     }
 
+    /**
+     * @Then I delete the :fullName address
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/AddressContext.php:99]:
```suggestion
     * @When I delete the :fullName address
```

File: features/account/customer_account/address_book/deleting_address_created_after_checkout.feature (hunk 3 above)
Comment (AdamKasp) [reply features/account/customer_account/address_book/deleting_address_created_after_checkout.feature]: And i think you can merge these two scenarios in one, you missed this part 💯

File: features/account/customer_account/address_book/deleting_address_created_after_checkout.feature (hunk 4 above)
Comment (arti0090) [reply features/account/customer_account/address_book/deleting_address_created_after_checkout.feature:30]: Mostly because step "And my billing address is fulfilled automatically through default address", as in API there is no such a think like automatically, but the step implementation we have now is a bit wrong. I think i will make refactor of Context ( or transformer) so it will be nice but in another PR.

File: src/Sylius/Behat/Context/Api/Shop/AddressContext.php [hunk 7]
```diff
@@ -111,4 +134,21 @@ private function addressBookHasAddress(array $addressBook, AddressInterface $add
...
         return false;
     }
+
+    private function getAddressIdFromAddressBookByFullName(string $fullName): ?string
+    {
+        Assert::notNull($fullName);
+        $fullNameArray = explode(' ',$fullName);
+
+        $addresses = $this->responseChecker->getCollection($this->client->getLastResponse());
+        /** @var AddressInterface $address */
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/AddressContext.php:144]: It is not that address I guess
```suggestion
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, @arti0090! :1st_place_medal:
